app/controllers/assessments_controller.rb

```diff
@@ -181,34 +181,46 @@ def update
 
     assessment = Assessment.find_by(id: params.permit(:id)[:id])
     reported_symptoms_array = params.permit({ symptoms: %i[name value type label notes required] }).to_h['symptoms']
+    redirect_to(root_url) && return if reported_symptoms_array.map { |symptom| symptom[:name] }.include?(:nil)
 
-    typed_reported_symptoms = Condition.build_symptoms(reported_symptoms_array)
+    symptom_types = assessment.reported_condition.threshold_condition.symptoms.pluck(:name, :label).to_h
```

Comment:
i don't think `symptom_types` accurately describes this. Maybe `symptom_names_to_labels` ? or something like that